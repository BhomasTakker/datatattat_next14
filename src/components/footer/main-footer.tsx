import React from "react";
import styles from "./main-footer.module.scss";
import { Icons } from "./icons";

const year = new Date(Date.now()).getFullYear();
const COPYRIGHT_NOTICE = `Copyright ${year} Â©Datatattat.`;
const COPYRIGHT_MESSAGE =
	"Datatattat is not responsible for the content of external sites.";

const Copyright = () => {
	return (
		<p className={styles.copyright}>
			<strong>{COPYRIGHT_NOTICE} </strong>
			{COPYRIGHT_MESSAGE}
		</p>
	);
};

export const MainFooter = () => {
	return (
		<footer className={styles.root}>
			<div className={styles.divider} />
			<div className={styles.container}>
				<Icons />
			</div>
			<div className={styles.divider} />
			<div className={styles.container}>
				<Copyright />
			</div>
		</footer>
	);
};
