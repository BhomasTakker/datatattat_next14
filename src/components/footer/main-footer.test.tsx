import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { MainFooter } from "./main-footer";

jest.mock("./icons", () => {
	return {
		Icons: () => <p>Icons</p>,
	};
});

describe("Main Footer", () => {
	it("renders footer", () => {
		render(<MainFooter />);
		const footer = screen.getByRole("contentinfo");
		expect(footer).toBeInTheDocument();
	});

	it("renders copyright notice", () => {
		render(<MainFooter />);
		const copyright = screen.getByText(
			/datatattat is not responsible for the content of external sites./i
		);
		expect(copyright).toBeInTheDocument();
	});

	it("renders current year", () => {
		render(<MainFooter />);
		const year = new Date(Date.now()).getFullYear();
		console.log({ year });
		// way 1 and 2
		// const message = screen.getByText(`${year}`, { exact: false });
		const message = screen.getByText(new RegExp(year.toString()));
		expect(message).toBeInTheDocument();
	});

	it("renders footer unchanged", () => {
		const { container } = render(<MainFooter />);
		expect(container).toMatchSnapshot();
	});
});
