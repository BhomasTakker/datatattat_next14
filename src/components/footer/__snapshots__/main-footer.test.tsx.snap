// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Main Footer renders footer unchanged 1`] = `
<div>
  <footer
    class="root"
  >
    <div
      class="divider"
    />
    <div
      class="container"
    >
      <p>
        Icons
      </p>
    </div>
    <div
      class="divider"
    />
    <div
      class="container"
    >
      <p
        class="copyright"
      >
        <strong>
          Copyright 2025 Â©Datatattat.
           
        </strong>
        Datatattat is not responsible for the content of external sites.
      </p>
    </div>
  </footer>
</div>
`;
